// bizcharts/lib/utils/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.shallowEqual = exports.cloneDeep = exports.visibleHelper = exports.splitBySeparator = exports.minifyNum = exports.percentage = exports.fold = void 0;
// 数据处理
var fold_1 = require("./data-transform/fold");
Object.defineProperty(exports, "fold", { enumerable: true, get: function () { return __importDefault(fold_1).default; } });
var percentage_1 = require("./data-transform/percentage");
Object.defineProperty(exports, "percentage", { enumerable: true, get: function () { return __importDefault(percentage_1).default; } });
var minifyNum_1 = require("./data-transform/minifyNum");
Object.defineProperty(exports, "minifyNum", { enumerable: true, get: function () { return __importDefault(minifyNum_1).default; } });
var splitBySeparator_1 = require("./data-transform/splitBySeparator");
Object.defineProperty(exports, "splitBySeparator", { enumerable: true, get: function () { return __importDefault(splitBySeparator_1).default; } });
// 配置项处理
var plotTools_1 = require("./plotTools");
Object.defineProperty(exports, "visibleHelper", { enumerable: true, get: function () { return plotTools_1.visibleHelper; } });
// 函数工具
var cloneDeep_1 = require("./cloneDeep");
Object.defineProperty(exports, "cloneDeep", { enumerable: true, get: function () { return __importDefault(cloneDeep_1).default; } });
var shallowEqual_1 = require("./shallowEqual");
Object.defineProperty(exports, "shallowEqual", { enumerable: true, get: function () { return __importDefault(shallowEqual_1).default; } });
